### PR TITLE
implement #28

### PR DESCRIPTION
**Pull request by ershovd**
Basic implementation of filtering incidents by`TaskStatus`.
Things to discuss:

- better default value for `ParseStatus`?
- should we filter by other tasks status except `TaskStatus.Incomplete`and `TaskStatus.UnableToComplete`
- maybe bind in event in XAML instead of IncidentCenterPage ctor?

### CODE REVIEW

File: QualityKitchen.Mobile/Application/ViewModels/IncidentCenterPageViewModel.cs [hunk 1]
```diff
@@ -33,10 +43,41 @@ public IncidentCenterPageViewModel ()
 			ShowOverlayCommand.Execute (null);
 
 			ServiceResult<List<DomainModel.Incident>> serviceResult = await App.Instance.IncidentService.GetAllIncidentsAsync (App.Instance.OperationId);
-			Incidents = new List<DomainModel.Incident> (serviceResult.Response);
-													
+
+			Incidents = new List<DomainModel.Incident>(serviceResult.Response);
+			_initialIncidetns = new List<DomainModel.Incident>(Incidents);
```
Comment (j0nscalet) [inline QualityKitchen.Mobile/Application/ViewModels/IncidentCenterPageViewModel.cs]: Fix spelling here variable name should be `_initialIncidents`

File: QualityKitchen.Mobile/Application/ViewModels/IncidentCenterPageViewModel.cs [hunk 2]
```diff
@@ -6,23 +6,33 @@
...
 
 			// TODO Clean up the MenuItem class								
 			this.SidebarItems = new List<MenuItem> () {
-				new MenuItem("Incidents",null,"All",15),
-				new MenuItem("Incidents",null,"Open",15),
-				new MenuItem("Incidents",null,"Blocked",15),
-				new MenuItem("Incidents",null,"Resolved",15),
-				new MenuItem("Incidents",null,"Closed",15)
-			};			
+				new MenuItem("Incidents",null,"All", menuItemBubbleCount),
```
Comment (j0nscalet) [inline QualityKitchen.Mobile/Application/ViewModels/IncidentCenterPageViewModel.cs]: My original code here was a bit funky. Should we parse the IncidentStatus enum to create the menu items?
This way if changes are made to those properties we don't have to circle back and add menu items.

What do you think?

File: QualityKitchen.Mobile/Application/ViewModels/IncidentCenterPageViewModel.cs [hunk 3]
```diff
@@ -6,23 +6,33 @@
...
+				new MenuItem("Incidents",null,"Open", menuItemBubbleCount),
+				new MenuItem("Incidents",null,"Blocked", menuItemBubbleCount),
+				new MenuItem("Incidents",null,"Resolved", menuItemBubbleCount),
+				new MenuItem("Incidents",null,"Closed", menuItemBubbleCount),
+				new MenuItem("Incidents",null,_flaggedMenuItemTitle, menuItemBubbleCount),
```
Comment (j0nscalet) [inline QualityKitchen.Mobile/Application/ViewModels/IncidentCenterPageViewModel.cs]: See comment below, you can use the string names like I'm do in line 31 and above. No need to create constant member variables.

File: QualityKitchen.Mobile/Application/ViewModels/IncidentCenterPageViewModel.cs [hunk 4]
```diff
@@ -33,10 +43,41 @@ public IncidentCenterPageViewModel ()
...
-			Incidents = new List<DomainModel.Incident> (serviceResult.Response);
-													
+
+			Incidents = new List<DomainModel.Incident>(serviceResult.Response);
+			_initialIncidetns = new List<DomainModel.Incident>(Incidents);	
+
 			HideOverlayCommand.Execute (null);
 		}
+
+		internal Domain.Model.Task.TaskStatus ParseStatus(MenuItem menu)
```
Comment (j0nscalet) [inline QualityKitchen.Mobile/Application/ViewModels/IncidentCenterPageViewModel.cs]: This ["smells"](https://en.wikipedia.org/wiki/Code_smell) to me, instead I think we should extend the MenuItem class and include something like a 'FriendlyName' property which could then be used by the UI when displaying the item.

Comment (ershovd) [inline QualityKitchen.Mobile/Application/ViewModels/IncidentCenterPageViewModel.cs]: @j0nscalet thats good idea, will do that.

**Comment by j0nscalet** [comment]
@ershovd we need to discuss, these changes didn't go that well.

**Pull request closed without merging**